src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = this.hidden
+      ? Boolean(0) // for linter's happiness)
+      : true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(victim) {
+    if (!(victim instanceof Carnivore) && !victim.hidden) {
+      victim.health -= 50;
+    }
+
+    if (victim.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(victim), 1);
```

Comment:
Perhaps, that's better to split finding an index and splicing into two different actions to reduce time complexity, isn't it?